maint/Makefile.am

```diff
@@ -7,11 +7,109 @@
 EXTRA_DIST		= README
 
-nodist_testcc_SOURCES	= testcc.cc
+nodist_testcc_common_acl_h_SOURCES = testcc_helper.cc
+testcc_common_acl_h_CXXFLAGS = -include "crm/common/acl.h"
+
+nodist_testcc_common_ipc_h_SOURCES = testcc_helper.cc
+testcc_common_ipc_h_CXXFLAGS = -include "crm/common/ipc.h"
+
+nodist_testcc_common_iso8601_h_SOURCES = testcc_helper.cc
+testcc_common_iso8601_h_CXXFLAGS = -include "crm/common/iso8601.h"
+
+nodist_testcc_common_mainloop_h_SOURCES = testcc_helper.cc
+testcc_common_mainloop_h_CXXFLAGS = -include "crm/common/mainloop.h"
+
+nodist_testcc_common_nvpair_h_SOURCES = testcc_helper.cc
+testcc_common_nvpair_h_CXXFLAGS = -include "crm/common/nvpair.h"
+
+nodist_testcc_common_results_h_SOURCES = testcc_helper.cc
+testcc_common_results_h_CXXFLAGS = -include "crm/common/results.h"
+
+nodist_testcc_common_util_h_SOURCES = testcc_helper.cc
+testcc_common_util_h_CXXFLAGS = -include "crm/common/util.h"
+
+nodist_testcc_common_xml_h_SOURCES = testcc_helper.cc
+testcc_common_xml_h_CXXFLAGS = -include "crm/common/xml.h"
+
+nodist_testcc_compatibility_h_SOURCES = testcc_helper.cc
+testcc_compatibility_h_CXXFLAGS = -include "crm/compatibility.h"
+
+nodist_testcc_pengine_common_h_SOURCES = testcc_helper.cc
+testcc_pengine_common_h_CXXFLAGS = -include "crm/pengine/common.h"
+
+nodist_testcc_pengine_complex_h_SOURCES = testcc_helper.cc
+testcc_pengine_complex_h_CXXFLAGS = -include "crm/pengine/complex.h"
+
+nodist_testcc_pe_types_h_SOURCES = testcc_helper.cc
+testcc_pe_types_h_CXXFLAGS = -include "crm/pengine/pe_types.h"
+
+nodist_testcc_cib_types_h_SOURCES = testcc_helper.cc
+testcc_cib_types_h_CXXFLAGS = -include "crm/cib/cib_types.h"
+
+nodist_testcc_cib_util_h_SOURCES = testcc_helper.cc
+testcc_cib_util_h_CXXFLAGS = -include "crm/cib/util.h"
+
+# the initial ones except election.h not packaged
+
+nodist_testcc_crm_h_SOURCES = testcc_helper.cc
+testcc_crm_h_CXXFLAGS = -include "crm/crm.h"
+
+nodist_testcc_cib_h_SOURCES = testcc_helper.cc
+testcc_cib_h_CXXFLAGS = -include "crm/cib.h"
+
+nodist_testcc_cluster_h_SOURCES = testcc_helper.cc
+testcc_cluster_h_CXXFLAGS = -include "crm/cluster.h"
+
+nodist_testcc_lrmd_h_SOURCES = testcc_helper.cc
+testcc_lrmd_h_CXXFLAGS = -include "crm/lrmd.h"
+
+nodist_testcc_msg_xml_h_SOURCES = testcc_helper.cc
+testcc_msg_xml_h_CXXFLAGS = -include "crm/msg_xml.h"
+
+nodist_testcc_services_h_SOURCES = testcc_helper.cc
+testcc_services_h_CXXFLAGS = -include "crm/services.h"
+
+nodist_testcc_stonith_ng_h_SOURCES = testcc_helper.cc
+testcc_stonith_ng_h_CXXFLAGS = -include "crm/stonith-ng.h"
+
+nodist_testcc_pengine_status_h_SOURCES = testcc_helper.cc
+testcc_pengine_status_h_CXXFLAGS = -include "crm/pengine/status.h"
+
+nodist_testcc_pengine_rules_h_SOURCES = testcc_helper.cc
+testcc_pengine_rules_h_CXXFLAGS = -include "crm/pengine/rules.h"
+
+# headers not pulling in everything they need
+
+nodist_testcc_common_logging_h_SOURCES = testcc_helper.cc
+testcc_common_logging_h_CXXFLAGS = -include "crm/common/logging.h"
```

Comment:
This feels like it will be difficult to maintain -- let's create a variable with a list of the header files, then use an implicit rule or loop to auto-generate everything else.

I just stumbled on this old note to myself that I had forgotten about: " see libqb/tests/makefile.am's "auto_check_header" for possibility to dynamically generate header-including test code (possibly in both C and C++)"

It's a good question how deep we should go with this. In the past there have been issues with certain combinations of headers, and even dependent on the order in which they were included (guard issues). Though that's applicable to C as well as C++.

FYI the testcc source rename is in the previous commit

Reply:
Well I needed something quick for finding the missing closing clause in acl.h (which I had overseen in my first grep :-( ). What was there didn't seem systematic (headers that are not packaged are checked while others that are packaged are not checked) so I looked over the headers we have and tried to group them. Not to loose that effort I put the result into that makefile. Guess the current state isn't really harder to maintain than what we had before.
I'd thus like to keep it like that for 2.0.4-rc1 and look into that automatism-stuff afterwards not to delay rc1 even further - especially as this is probably not interfering with others testing out rc1. I'll move the renaming to the same commit.

Reply:
Makes sense